File: tasking/sources/task_dep.3.c
/*
* @@name:	task_dep.3
* @@type:	C
* @@operation:	run
* @@expect:	success
* @@version:	omp_4.0
*/
#include <stdio.h>
int main() {
   int x;
   #pragma omp parallel
   #pragma omp single
   {
      #pragma omp task shared(x) depend(out: x)
         x = 1;
      #pragma omp task shared(x) depend(out: x)
         x = 2;
      #pragma omp taskwait
      printf("x = %d\n", x);
   }
   return 0;
}

File: tasking/sources/task_dep.3.f90
! @@name:	task_dep.3
! @@type:	F-free
! @@operation:	run
! @@expect:	success
! @@version:	omp_4.0
program example
   integer :: x
   !$omp parallel
   !$omp single
      !$omp task shared(x) depend(out: x)
         x = 1
      !$omp end task
      !$omp task shared(x) depend(out: x)
         x = 2
      !$omp end task
      !$omp taskwait
      print*, "x = ", x
   !$omp end single
   !$omp end parallel
end program
